src/Infrastructure/Stize.Infrastructure.Azure/Networking/SecurityRuleExtensions.cs

```diff
@@ -0,0 +1,222 @@
+﻿using Pulumi;
+using Pulumi.AzureNextGen.Network.Latest;
+using Pulumi.AzureNextGen.Resources.Latest;
+using Inputs = Pulumi.AzureNextGen.Network.Latest.Inputs;
+
+namespace Stize.Infrastructure.Azure.Networking
+{
+    public static class SecurityRuleExtensions
+    {
+        /// <summary>
+        /// Sets the builder name. If the builder has an RandomId associated, 
+        /// appends the hex value of the RandomId to the end of the name
+        /// </summary>
+        /// <param name="builder"><see cref="SecurityRule"/> builder</param>
+        /// <param name="name"><see cref="SecurityRule"/> name</param>
+        /// <returns></returns>
+        public static SecurityRuleBuilder Name(this SecurityRuleBuilder builder, Input<string> name)
+        {
+            if (builder.RandomId != null)
+            {
+                builder.Arguments.SecurityRuleName = builder.RandomId.Hex.Apply(r => $"{name}-{r}");
```

Comment:
And here. I recently found that the logic for creating the random names was wrong. Please find a corrected example:

https://github.com/stize/infrastructure/blob/0122c548778a1a6789fd91ca8b3368be2f6af32b/src/Infrastructure/Stize.Infrastructure.Azure/ResourceGroupExtensions.cs#L18

Use the same approach here please.

The reason being, name and Hex are both Pulumi Input values. We need to Apply both (in a concatenation) in order for the evaluated values to be passed to the resource name.